src/instructlab/data/generate.py

```diff
@@ -155,6 +156,12 @@
     is_flag=True,
     help="Print serving engine logs.",
 )
+@click.option(
+    "--gpus",
+    type=click.STRING,
```

Comment:
Should this be an INT or click.IntRange(min=0)

Reply:
can't GPUs be the following format:

`0-4`
`0-2,4,8`
`ALL`
?

Reply:
I am not seeing that in the vllm docs.  Do you have a reference?  Two other things to think about:

- Are we wanting to pass through vllm specific args or are we going to translate user input for --gpus to whatever backend they picked.  Understanding it's only applicable to vllm today.
- We should be consistent with evaluate whatever we pick

Reply:
I am not seeing that in the vllm docs. Do you have a reference? Two other things to think about:

Are we wanting to pass through vllm specific args or are we going to translate user input for --gpus to whatever backend they picked. Understanding it's only applicable to vllm today.
We should be consistent with evaluate whatever we pick

Reply:
I think for now, `--gpus` should only be a translation into vllm. You are right, I was assuming it follows the standards "set" notation for GPU and CPU flags but it seems to be only an int for vllm